diff --git a/docs/docs.go b/docs/docs.go
index bf84a22..6973c04 100644
--- a/docs/docs.go
+++ b/docs/docs.go
@@ -387,6 +387,61 @@ const docTemplate = `{
                 }
             }
         },
+        "/files/{name}": {
+            "get": {
+                "description": "Download a file by its name. Returns file content as attachment.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/octet-stream",
+                    "image/jpeg",
+                    "image/png",
+                    "application/pdf",
+                    "text/plain"
+                ],
+                "tags": [
+                    "files"
+                ],
+                "summary": "Download file by name",
+                "parameters": [
+                    {
+                        "type": "string",
+                        "description": "File name",
+                        "name": "name",
+                        "in": "path",
+                        "required": true
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "File content",
+                        "schema": {
+                            "type": "file"
+                        },
+                        "headers": {
+                            "Content-Disposition": {
+                                "type": "string",
+                                "description": "attachment; filename=example.jpg"
+                            },
+                            "Content-Type": {
+                                "type": "string",
+                                "description": "MIME type of the file"
+                            }
+                        }
+                    },
+                    "500": {
+                        "description": "Internal server error",
+                        "schema": {
+                            "type": "object",
+                            "additionalProperties": {
+                                "type": "string"
+                            }
+                        }
+                    }
+                }
+            }
+        },
         "/users": {
             "get": {
                 "description": "Get all users",
diff --git a/docs/swagger.json b/docs/swagger.json
index ba826d1..d336177 100644
--- a/docs/swagger.json
+++ b/docs/swagger.json
@@ -381,6 +381,61 @@
                 }
             }
         },
+        "/files/{name}": {
+            "get": {
+                "description": "Download a file by its name. Returns file content as attachment.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/octet-stream",
+                    "image/jpeg",
+                    "image/png",
+                    "application/pdf",
+                    "text/plain"
+                ],
+                "tags": [
+                    "files"
+                ],
+                "summary": "Download file by name",
+                "parameters": [
+                    {
+                        "type": "string",
+                        "description": "File name",
+                        "name": "name",
+                        "in": "path",
+                        "required": true
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "File content",
+                        "schema": {
+                            "type": "file"
+                        },
+                        "headers": {
+                            "Content-Disposition": {
+                                "type": "string",
+                                "description": "attachment; filename=example.jpg"
+                            },
+                            "Content-Type": {
+                                "type": "string",
+                                "description": "MIME type of the file"
+                            }
+                        }
+                    },
+                    "500": {
+                        "description": "Internal server error",
+                        "schema": {
+                            "type": "object",
+                            "additionalProperties": {
+                                "type": "string"
+                            }
+                        }
+                    }
+                }
+            }
+        },
         "/users": {
             "get": {
                 "description": "Get all users",
diff --git a/docs/swagger.yaml b/docs/swagger.yaml
index 36029a8..c20e544 100644
--- a/docs/swagger.yaml
+++ b/docs/swagger.yaml
@@ -326,6 +326,44 @@ paths:
       summary: VerifyCode
       tags:
       - Auth
+  /files/{name}:
+    get:
+      consumes:
+      - application/json
+      description: Download a file by its name. Returns file content as attachment.
+      parameters:
+      - description: File name
+        in: path
+        name: name
+        required: true
+        type: string
+      produces:
+      - application/octet-stream
+      - image/jpeg
+      - image/png
+      - application/pdf
+      - text/plain
+      responses:
+        "200":
+          description: File content
+          headers:
+            Content-Disposition:
+              description: attachment; filename=example.jpg
+              type: string
+            Content-Type:
+              description: MIME type of the file
+              type: string
+          schema:
+            type: file
+        "500":
+          description: Internal server error
+          schema:
+            additionalProperties:
+              type: string
+            type: object
+      summary: Download file by name
+      tags:
+      - files
   /files/upload-temporary:
     post:
       consumes:
diff --git a/internal/app/provider.go b/internal/app/provider.go
index 01b0598..94dcaa1 100644
--- a/internal/app/provider.go
+++ b/internal/app/provider.go
@@ -64,8 +64,8 @@ func (p *moduleProvider) AuthModule() error {
 		p.app.db,
 		p.app.redisManager,
 		p.app.config,
-		p.userModule.UserUsecase,
-		p.notificationModule.Service,
+		p.userModule.GetUserUsecase(),
+		p.notificationModule.GetNotificationService(),
 	)
 	p.authModule.Init()
 	return nil
diff --git a/internal/config/config.go b/internal/config/config.go
index 86623fe..9d8ecc5 100644
--- a/internal/config/config.go
+++ b/internal/config/config.go
@@ -16,6 +16,8 @@ type Config struct {
 	RedisUrl            string        `mapstructure:"REDIS_URL"`
 	VerifyCodeExpiredIn time.Duration `mapstructure:"VERIFY_CODE_EXPIRED_IN"`
 	FileDir             string        `mapstructure:"FILE_DIR"`
+	ApiUrl              string        `mapstructure:"API_URL"`
+	FileLink            string        `mapstructure:"FILE_LINK"`
 
 	CorsOrigin string `mapstructure:"CORS_ORIGIN"`
 
diff --git a/internal/module/file/delivery/http/handler.go b/internal/module/file/delivery/http/handler.go
index ab16703..5dbf4c7 100644
--- a/internal/module/file/delivery/http/handler.go
+++ b/internal/module/file/delivery/http/handler.go
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	file_entity "github.com/Fi44er/sdmed/internal/module/file/entity"
+	"github.com/Fi44er/sdmed/internal/module/file/pkg/utils"
 	"github.com/Fi44er/sdmed/pkg/logger"
 	_ "github.com/Fi44er/sdmed/pkg/response"
 	"github.com/go-playground/validator/v10"
@@ -13,7 +14,8 @@ import (
 )
 
 type IFileUsecase interface {
-	UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) error
+	UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) (string, error)
+	Get(ctx context.Context, name string) (*file_entity.File, error)
 }
 
 type FileHandler struct {
@@ -75,7 +77,8 @@ func (h *FileHandler) UploadTemporary(ctx *fiber.Ctx) error {
 		Data: fileData,
 	}
 
-	if err := h.usecase.UploadTemporary(ctx.Context(), &entity, 5*time.Second); err != nil {
+	link, err := h.usecase.UploadTemporary(ctx.Context(), &entity, 10*time.Minute)
+	if err != nil {
 		return ctx.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
 			"error": err.Error(),
 		})
@@ -83,5 +86,40 @@ func (h *FileHandler) UploadTemporary(ctx *fiber.Ctx) error {
 
 	return ctx.Status(fiber.StatusCreated).JSON(fiber.Map{
 		"message": "File uploaded successfully",
+		"url":     link,
 	})
 }
+
+// Get download file by name
+// @Summary Download file by name
+// @Description Download a file by its name. Returns file content as attachment.
+// @Tags files
+// @Accept json
+// @Produce application/octet-stream
+// @Produce image/jpeg
+// @Produce image/png
+// @Produce application/pdf
+// @Produce text/plain
+// @Param name path string true "File name"
+// @Success 200 {file} byte "File content"
+// @Header 200 {string} Content-Disposition "attachment; filename=example.jpg"
+// @Header 200 {string} Content-Type "MIME type of the file"
+// @Failure 500 {object} map[string]string "Internal server error"
+// @Router /files/{name} [get]
+func (h *FileHandler) Get(ctx *fiber.Ctx) error {
+	name := ctx.Params("name")
+
+	file, err := h.usecase.Get(ctx.Context(), name)
+	if err != nil {
+		return ctx.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": err.Error(),
+		})
+	}
+	ctx.Set("Content-Disposition", "inline; filename="+file.Name)
+	ctx.Set("Content-Type", "application/octet-stream")
+
+	contentType := utils.GetContentType(file.Name)
+	ctx.Set("Content-Type", contentType)
+
+	return ctx.Send(file.Data)
+}
diff --git a/internal/module/file/delivery/http/route.go b/internal/module/file/delivery/http/route.go
index adfda73..5e53fc7 100644
--- a/internal/module/file/delivery/http/route.go
+++ b/internal/module/file/delivery/http/route.go
@@ -6,4 +6,5 @@ func (h *FileHandler) RegisterRoutes(router fiber.Router) {
 	files := router.Group("/files")
 
 	files.Post("/upload-temporary", h.UploadTemporary)
+	files.Get("/:name", h.Get)
 }
diff --git a/internal/module/file/module.go b/internal/module/file/module.go
index b27e881..af91c53 100644
--- a/internal/module/file/module.go
+++ b/internal/module/file/module.go
@@ -52,9 +52,9 @@ func (m *FileModule) Init() {
 
 	m.fileRepository = repository.NewFileRepository(m.logger, m.db)
 	m.fileStorage = filesystem.NewLocalFileStorage(m.logger, m.config)
-	m.fileUsecase = file_usecase.NewFileUsecase(m.fileRepository, m.uow, m.fileStorage, m.logger)
+	m.fileUsecase = file_usecase.NewFileUsecase(m.fileRepository, m.uow, m.fileStorage, m.logger, m.config)
 	m.fileHandler = file_http.NewFileHandler(m.fileUsecase, m.validator, m.logger)
-	m.fileCleaner = file_usecase.NewFileCleaner(m.fileRepository, m.fileStorage, m.logger, 10*time.Second, 10*time.Second)
+	m.fileCleaner = file_usecase.NewFileCleaner(m.fileRepository, m.fileStorage, m.logger, 20*time.Minute)
 }
 
 func (m *FileModule) InitDelivery(router fiber.Router) {
diff --git a/internal/module/file/usecase/file/cleaner.go b/internal/module/file/usecase/file/cleaner.go
index 0ff2473..c8ac60e 100644
--- a/internal/module/file/usecase/file/cleaner.go
+++ b/internal/module/file/usecase/file/cleaner.go
@@ -14,7 +14,6 @@ type FileCleaner struct {
 	fileStorage IFileStorage
 	logger      *logger.Logger
 	interval    time.Duration
-	ttl         time.Duration
 	stopCh      chan struct{}
 	running     bool
 	mutex       sync.RWMutex
@@ -29,14 +28,12 @@ func NewFileCleaner(
 	fileStorage IFileStorage,
 	logger *logger.Logger,
 	interval time.Duration,
-	ttl time.Duration,
 ) *FileCleaner {
 	return &FileCleaner{
 		repository:  repository,
 		fileStorage: fileStorage,
 		logger:      logger,
 		interval:    interval,
-		ttl:         ttl,
 		stopCh:      make(chan struct{}),
 	}
 }
diff --git a/internal/module/file/usecase/file/usecase.go b/internal/module/file/usecase/file/usecase.go
index 1d9de4a..9161625 100644
--- a/internal/module/file/usecase/file/usecase.go
+++ b/internal/module/file/usecase/file/usecase.go
@@ -2,8 +2,10 @@ package file_usecase
 
 import (
 	"context"
+	"fmt"
 	"time"
 
+	"github.com/Fi44er/sdmed/internal/config"
 	file_entity "github.com/Fi44er/sdmed/internal/module/file/entity"
 	"github.com/Fi44er/sdmed/pkg/logger"
 	"github.com/Fi44er/sdmed/pkg/postgres/uow"
@@ -29,12 +31,14 @@ type FileUsecase struct {
 	uow         uow.Uow
 	fileStorage IFileStorage
 	logger      *logger.Logger
+	config      *config.Config
 }
 
 type IFileUsecase interface {
-	UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) error
-	UploadPermanent(ctx context.Context, file *file_entity.File, ownerID, ownerType string) error
+	UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) (string, error)
+	UploadPermanent(ctx context.Context, file *file_entity.File, ownerID, ownerType string) (string, error)
 	MakeFilesPermanent(ctx context.Context, fileIDs []string, ownerID, ownerType string) error
+	Get(ctx context.Context, name string) (*file_entity.File, error)
 }
 
 func NewFileUsecase(
@@ -42,26 +46,28 @@ func NewFileUsecase(
 	uow uow.Uow,
 	fileStorage IFileStorage,
 	logger *logger.Logger,
+	config *config.Config,
 ) IFileUsecase {
 	return &FileUsecase{
 		repository:  repository,
 		uow:         uow,
 		fileStorage: fileStorage,
 		logger:      logger,
+		config:      config,
 	}
 }
 
-func (u *FileUsecase) UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) error {
+func (u *FileUsecase) UploadTemporary(ctx context.Context, file *file_entity.File, ttl time.Duration) (string, error) {
 	return u.uploadWithStatus(ctx, file, file_entity.FileStatusTemporary, ttl)
 }
 
-func (u *FileUsecase) UploadPermanent(ctx context.Context, file *file_entity.File, ownerID, ownerType string) error {
-	if err := u.uploadWithStatus(ctx, file, file_entity.FileStatusPermanent, 0); err != nil {
-		return err
+func (u *FileUsecase) UploadPermanent(ctx context.Context, file *file_entity.File, ownerID, ownerType string) (string, error) {
+	url, err := u.uploadWithStatus(ctx, file, file_entity.FileStatusPermanent, 0)
+	if err != nil {
+		return "", err
 	}
 
-	// Обновляем файл как постоянный
-	return u.uow.Do(ctx, func(ctx context.Context) error {
+	err = u.uow.Do(ctx, func(ctx context.Context) error {
 		repo, err := u.uow.GetRepository(ctx, "file")
 		if err != nil {
 			return err
@@ -71,6 +77,8 @@ func (u *FileUsecase) UploadPermanent(ctx context.Context, file *file_entity.Fil
 		file.MarkAsPermanent(ownerID, ownerType)
 		return fileRepo.Update(ctx, file)
 	})
+
+	return url, err
 }
 
 func (u *FileUsecase) MakeFilesPermanent(ctx context.Context, fileIDs []string, ownerID, ownerType string) error {
@@ -96,14 +104,14 @@ func (u *FileUsecase) MakeFilesPermanent(ctx context.Context, fileIDs []string,
 	})
 }
 
-func (u *FileUsecase) uploadWithStatus(ctx context.Context, file *file_entity.File, status file_entity.FileStatus, ttl time.Duration) error {
-	return u.uow.Do(ctx, func(ctx context.Context) error {
+func (u *FileUsecase) uploadWithStatus(ctx context.Context, file *file_entity.File, status file_entity.FileStatus, ttl time.Duration) (string, error) {
+	var fileName string
+	err := u.uow.Do(ctx, func(ctx context.Context) error {
 		if err := file.GenerateName(); err != nil {
 			u.logger.Errorf("failed to generate file name: %s", err)
 			return err
 		}
 
-		// Устанавливаем статус и TTL
 		file.Status = status
 		if status == file_entity.FileStatusTemporary && ttl > 0 {
 			expiresAt := time.Now().Add(ttl)
@@ -123,6 +131,8 @@ func (u *FileUsecase) uploadWithStatus(ctx context.Context, file *file_entity.Fi
 			return err
 		}
 
+		fileName = file.Name
+
 		repo, err := u.uow.GetRepository(ctx, "file")
 		if err != nil {
 			return err
@@ -136,4 +146,22 @@ func (u *FileUsecase) uploadWithStatus(ctx context.Context, file *file_entity.Fi
 		needCleanup = false
 		return nil
 	})
+
+	link := fmt.Sprintf("%s/%s/%s", u.config.ApiUrl, u.config.FileLink, fileName)
+
+	return link, err
+}
+
+func (u *FileUsecase) Get(ctx context.Context, name string) (*file_entity.File, error) {
+	file, err := u.repository.GetByName(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := u.fileStorage.Get(name)
+	if err != nil {
+		return nil, err
+	}
+	file.Data = data
+	return file, nil
 }
diff --git a/internal/module/notification/module.go b/internal/module/notification/module.go
index b1e4ba9..9686503 100644
--- a/internal/module/notification/module.go
+++ b/internal/module/notification/module.go
@@ -11,7 +11,7 @@ type NotificationModule struct {
 	logger *logger.Logger
 	config *config.Config
 
-	Service *service.NotificationService
+	service *service.NotificationService
 }
 
 func NewNotificationModule(logger *logger.Logger, config *config.Config) *NotificationModule {
@@ -38,7 +38,11 @@ func (m *NotificationModule) Init() error {
 		"smtp": smtp,
 	}
 
-	m.Service = service.NewNotificationService(notifiers, m.logger)
+	m.service = service.NewNotificationService(notifiers, m.logger)
 
 	return nil
 }
+
+func (m *NotificationModule) GetNotificationService() *service.NotificationService {
+	return m.service
+}
diff --git a/internal/module/product/dto/category.go b/internal/module/product/dto/category.go
index 7395f6d..a523530 100644
--- a/internal/module/product/dto/category.go
+++ b/internal/module/product/dto/category.go
@@ -3,3 +3,8 @@ package product_dto
 type CategoryDTO struct {
 	Name string `json:"name"`
 }
+
+type CreateCategoryDTO struct {
+	Name   string `json:"name"`
+	Images []File
+}
diff --git a/internal/module/product/entity/file.go b/internal/module/product/entity/file.go
index b7bfeb6..ac24510 100644
--- a/internal/module/product/entity/file.go
+++ b/internal/module/product/entity/file.go
@@ -3,7 +3,7 @@ package product_entity
 type File struct {
 	ID        string
 	Name      string
-	OwnerID   string
-	OwnerType string
 	Data      []byte
+	OwnerID   *string
+	OwnerType *string
 }
diff --git a/internal/module/product/infrastucture/adapters/file_usecase.go b/internal/module/product/infrastucture/adapters/file_usecase.go
index 8f74d44..19881b3 100644
--- a/internal/module/product/infrastucture/adapters/file_usecase.go
+++ b/internal/module/product/infrastucture/adapters/file_usecase.go
@@ -1,10 +1,15 @@
 package product_adapters
 
 import (
+	"context"
+
+	file_entity "github.com/Fi44er/sdmed/internal/module/file/entity"
 	file_usecase "github.com/Fi44er/sdmed/internal/module/file/usecase/file"
+	product_entity "github.com/Fi44er/sdmed/internal/module/product/entity"
 )
 
 type IFileUsecaseAdapter interface {
+	MakeFilesPermanent(ctx context.Context, fileIDs []string, ownerID, ownerType string) error
 	// UploadFile(ctx context.Context, file *product_entity.File) error
 	// GetFile(ctx context.Context, name string) (*product_entity.File, error)
 }
@@ -19,6 +24,10 @@ func NewFileUsecaseAdapter(fileUsecase file_usecase.IFileUsecase) IFileUsecaseAd
 	}
 }
 
+func (a *FileUsecaseAdapter) MakeFilesPermanent(ctx context.Context, fileIDs []string, ownerID, ownerType string) error {
+	return a.fileUsecase.MakeFilesPermanent(ctx, fileIDs, ownerID, ownerType)
+}
+
 // func (a *FileUsecaseAdapter) UploadFile(ctx context.Context, file *product_entity.File) error {
 // 	fileproduct_entity := toFile(file)
 // 	return a.fileUsecase.Upload(ctx, fileproduct_entity)
@@ -29,15 +38,15 @@ func NewFileUsecaseAdapter(fileUsecase file_usecase.IFileUsecase) IFileUsecaseAd
 // 	return toProductFile(file), err
 // }
 
-// func toFile(file *product_entity.File) *file_entity.File {
-// 	return &file_entity.File{
-// 		ID:        file.ID,
-// 		Name:      file.Name,
-// 		OwnerID:   file.OwnerID,
-// 		OwnerType: file.OwnerType,
-// 		Data:      file.Data,
-// 	}
-// }
+func toFile(file *product_entity.File) *file_entity.File {
+	return &file_entity.File{
+		ID:        file.ID,
+		Name:      file.Name,
+		OwnerID:   file.OwnerID,
+		OwnerType: file.OwnerType,
+		Data:      file.Data,
+	}
+}
 
 // func toProductFile(file *file_entity.File) *product_entity.File {
 // 	return &product_entity.File{
diff --git a/internal/module/product/usecase/category/usecase.go b/internal/module/product/usecase/category/usecase.go
index abdefbb..3bffc3b 100644
--- a/internal/module/product/usecase/category/usecase.go
+++ b/internal/module/product/usecase/category/usecase.go
@@ -10,8 +10,7 @@ import (
 )
 
 type IFileUsecaseAdapter interface {
-	// UploadFile(ctx context.Context, file *product_entity.File) error
-	// GetFile(ctx context.Context, name string) (*product_entity.File, error)
+	MakeFilesPermanent(ctx context.Context, fileIDs []string, ownerID, ownerType string) error
 }
 
 type ICategoryUsecase interface {
diff --git a/internal/module/user/module.go b/internal/module/user/module.go
index 2cd9124..8d54531 100644
--- a/internal/module/user/module.go
+++ b/internal/module/user/module.go
@@ -13,7 +13,7 @@ import (
 
 type UserModule struct {
 	userRepository *user_repository.UserRepository
-	UserUsecase    *user_usecase.UserUsecase
+	userUsecase    *user_usecase.UserUsecase
 	userHandler    *user_handler.UserHandler
 
 	logger    *logger.Logger
@@ -35,10 +35,14 @@ func NewUserModule(
 
 func (m *UserModule) Init() {
 	m.userRepository = user_repository.NewUserRepository(m.logger, m.db)
-	m.UserUsecase = user_usecase.NewUserUsecase(m.userRepository, m.logger)
-	m.userHandler = user_handler.NewUserHandler(m.UserUsecase, m.logger, m.validator)
+	m.userUsecase = user_usecase.NewUserUsecase(m.userRepository, m.logger)
+	m.userHandler = user_handler.NewUserHandler(m.userUsecase, m.logger, m.validator)
 }
 
 func (m *UserModule) InitDelivery(router fiber.Router) {
 	m.userHandler.RegisterRoutes(router)
 }
+
+func (m *UserModule) GetUserUsecase() *user_usecase.UserUsecase {
+	return m.userUsecase
+}
diff --git a/internal/module/user/usecase/user/usecase.go b/internal/module/user/usecase/user/usecase.go
index 2f4e19b..b644dd1 100644
--- a/internal/module/user/usecase/user/usecase.go
+++ b/internal/module/user/usecase/user/usecase.go
@@ -3,8 +3,8 @@ package user_usecase
 import (
 	"context"
 
-	"github.com/Fi44er/sdmed/internal/module/user/entity"
-	"github.com/Fi44er/sdmed/internal/module/user/pkg/constant"
+	user_entity "github.com/Fi44er/sdmed/internal/module/user/entity"
+	user_constant "github.com/Fi44er/sdmed/internal/module/user/pkg/constant"
 	"github.com/Fi44er/sdmed/pkg/logger"
 )
 
